cmd/common/utils.go

```diff
@@ -0,0 +1,93 @@
+
+	csiconfig "sigs.k8s.io/vsphere-csi-driver/pkg/common/config"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common/commonco/k8sorchestrator"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+func SetInitParams(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor, initParams *interface{},
+supervisorFSSName, supervisorFSSNamespace, internalFSSName, internalFSSNamespace string) {
+	log := logger.GetLogger(ctx)
+	// Set default values for FSS, if not given and initiate CO-agnostic init params
+	switch clusterFlavor {
+	case cnstypes.CnsClusterFlavorWorkload:
+		if strings.TrimSpace(supervisorFSSName) == "" {
+			log.Infof("Defaulting feature states configmap name to %q", csiconfig.DefaultSupervisorFSSConfigMapName)
+			supervisorFSSName = csiconfig.DefaultSupervisorFSSConfigMapName
+		}
+		if strings.TrimSpace(supervisorFSSNamespace) == "" {
+			log.Infof("Defaulting feature states configmap namespace to %q", csiconfig.DefaultCSINamespace)
+			supervisorFSSNamespace = csiconfig.DefaultCSINamespace
+		}
+		*initParams = k8sorchestrator.K8sSupervisorInitParams{
+			SupervisorFeatureStatesConfigInfo: csiconfig.FeatureStatesConfigInfo{
+				Name:      supervisorFSSName,
+				Namespace: supervisorFSSNamespace,
+			},
+		}
+	case cnstypes.CnsClusterFlavorVanilla:
+		if strings.TrimSpace(internalFSSName) == "" {
+			log.Infof("Defaulting feature states configmap name to %q", csiconfig.DefaultInternalFSSConfigMapName)
+			internalFSSName = csiconfig.DefaultInternalFSSConfigMapName
+		}
+		if strings.TrimSpace(internalFSSNamespace) == "" {
+			log.Infof("Defaulting feature states configmap namespace to %q", csiconfig.DefaultCSINamespaceVanillaK8s)
+			internalFSSNamespace = csiconfig.DefaultCSINamespaceVanillaK8s
+		}
+		*initParams = k8sorchestrator.K8sVanillaInitParams{
+			InternalFeatureStatesConfigInfo: csiconfig.FeatureStatesConfigInfo{
+				Name:      internalFSSName,
+				Namespace: internalFSSNamespace,
+			},
+		}
+	case cnstypes.CnsClusterFlavorGuest:
+		if strings.TrimSpace(supervisorFSSName) == "" {
+			log.Infof("Defaulting supervisor feature states configmap name to %q", csiconfig.DefaultSupervisorFSSConfigMapName)
+			supervisorFSSName = csiconfig.DefaultSupervisorFSSConfigMapName
+		}
+		if strings.TrimSpace(supervisorFSSNamespace) == "" {
+			log.Infof("Defaulting supervisor feature states configmap namespace to %q", csiconfig.DefaultCSINamespace)
+			supervisorFSSNamespace = csiconfig.DefaultCSINamespace
+		}
+		if strings.TrimSpace(internalFSSName) == "" {
+			log.Infof("Defaulting internal feature states configmap name to %q", csiconfig.DefaultInternalFSSConfigMapName)
+			internalFSSName = csiconfig.DefaultInternalFSSConfigMapName
+		}
+		if strings.TrimSpace(internalFSSNamespace) == "" {
+			log.Infof("Defaulting internal feature states configmap namespace to %q", csiconfig.DefaultCSINamespace)
+			internalFSSNamespace = csiconfig.DefaultCSINamespace
+		}
+		*initParams = k8sorchestrator.K8sGuestInitParams{
```

Comment:
Can you print `initParams` with debug log?

Reply:
Done